tensorflow_lite_support/c/task/vision/image_classifier.h

```diff
@@ -0,0 +1,189 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+#ifndef TENSORFLOW_LITE_SUPPORT_C_TASK_VISION_IMAGE_CLASSIFIER_H_
+#define TENSORFLOW_LITE_SUPPORT_C_TASK_VISION_IMAGE_CLASSIFIER_H_
+
+#include <stdint.h>
+
+#include "tensorflow_lite_support/c/task/vision/core/frame_buffer.h"
+#include "tensorflow_lite_support/c/task/processor/classification_result.h"
+#include "tensorflow_lite_support/c/task/vision/vision_common.h"
+
+// --------------------------------------------------------------------------
+/// C API for ImageClassifiier.
+///
+/// The API leans towards simplicity and uniformity instead of convenience, as
+/// most usage will be by language-specific wrappers. It provides largely the
+/// same set of functionality as that of the C++ TensorFlow Lite
+/// `ImageClassifier` API, but is useful for shared libraries where having
+/// a stable ABI boundary is important.
+///
+/// Usage with Model File Path:
+/// <pre><code>
+/// // Create the model
+/// ImageClassifier* classifier =
+///   ImageClassifierFromFile("/path/to/model.tflite");
+///
+/// struct FrameBuffer frame_buffer = { Initialize }
+///
+/// Create the model
+/// struct ClassificationResult *classification_result = 
+///   ImageClassifierClassify(classifier, &frame_buffer,);
+///
+/// // Dispose of the API object.
+/// ImageClassifierDelete(classifier);
+/// 
+/// Advanced Usage with Options:
+/// <pre><code>
+/// // Create the interpreter options
+/// ImageClassifierOptions *options = ImageClassifierOptionsCreate();
```

Comment:
@khanhlvg I initially used a simple struct but decided against it later. I had shared a doc when I implemented this with my reasoning. Image Classifier options have arrays like class name whitelist and black list which are of unknown sizes. We can't set the pointer to the base address of these arrays in the respective C++ options. We have to loop through them and add them into the respective C++ options in the proto buf. For that we need the length of these arrays. If we are going with the struct approach we will have to add members for the sizes of these options. Another way was to let users of the C apis add options via methods. Got the idea from [TfLiteInterpreterOptions](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/c/c_api.h)
I felt the latter approach is better since it eliminates addition of unwanted size members in the struct. 
A different strategy would be to have a nested struct type like say struct StringArrayOption within Classifier Options with size and pointer to array as members and let all string array options be of that type. If that's okay we can change this. Or if you have another workaround or approach, please let me know.

Reply:
As Lu's suggestion in her comment in the doc, let's create a TfLiteStringOption struct that contains an array and an int to store the array size.

Reply:
Implemented